sdk/storage/azure-storage-file/src/test/java/com/azure/storage/file/spock/FileServiceAPITests.groovy

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.spock
+
+import com.azure.storage.common.credentials.SharedKeyCredential
+import com.azure.storage.file.ShareClient
+import com.azure.storage.file.models.CorsRule
+import com.azure.storage.file.models.FileServiceProperties
+import com.azure.storage.file.models.ListSharesOptions
+import com.azure.storage.file.models.Metrics
+import com.azure.storage.file.models.RetentionPolicy
+import com.azure.storage.file.models.ShareItem
+import com.azure.storage.file.models.ShareProperties
+import com.azure.storage.file.models.StorageErrorCode
+import com.azure.storage.file.models.StorageErrorException
+import spock.lang.Unroll
+
+import static org.junit.Assert.assertFalse
+
+class FileServiceAPITests extends APISpec {
+    def shareName
+
+    static def testMetadata = Collections.singletonMap("testmetadata", "value")
+    static def reallyLongString = "thisisareallylongstringthatexceedsthe64characterlimitallowedoncertainproperties"
+    static def TOO_MANY_RULES = new ArrayList<>()
+    static def INVALID_ALLOWED_HEADER = Collections.singletonList(new CorsRule().allowedHeaders(reallyLongString))
```

Comment:
These setups may be wrong given that all properties of the `CorsRule` must be present when setting the rule. These might be testing the wrong piece of the service.

https://docs.microsoft.com/en-us/rest/api/storageservices/set-file-service-properties#request-body